backend/fms_core/resources/experiment_run.py

```diff
@@ -91,42 +91,41 @@ def before_import(self, dataset, using_transactions, dry_run, **kwargs):
         for protocol_name in protocols_for_experiment_type.keys():
             p = Protocol.objects.get(name=protocol_name)
             subprotocol_names = protocols_for_experiment_type[protocol_name]
-            self.protocols_dict[p] = map(lambda x: Protocol.objects.get(name=x), subprotocol_names)
+            subprotocols = []
+            for subprotocol_name in subprotocol_names:
+               subprotocols.append(Protocol.objects.get(name=subprotocol_name))
+            self.protocols_dict[p] = subprotocols.copy()
 
 
         skip_rows(dataset, 10)  # Skip preamble
 
 
-    def before_import_row(self, row, **kwargs):
-        self.temporary_experiment_id = row.get("Experiment ID")
-        super().before_import_row(row, **kwargs)
-
-
     def import_row(self, row, instance_loader, using_transactions=True, dry_run=False, raise_errors=False, **kwargs):
-        import_result = super().import_row(row, instance_loader, using_transactions, dry_run, raise_errors, **kwargs)
-
-        row_id = row.get("#")
-
-        # We are in the Sample section
-        # Columns from the Experiment section are mapped to the ones in the Sample section
-        # This is hacky, and has to be replaced when we use a different tool for importing the data
-
-
-        if self.is_in_samples_section:
-            sample_row = {
-                "#": row.get("#"),
-                "experiment_id": row.get("Experiment ID"),
-                "source_container_barcode": row.get("Experiment Container Barcode"),
-                "source_container_position": row.get("Experiment Container Kind"),
-                "source_sample_volume_used": float_to_decimal(blank_str_to_none(row.get("Instrument Name"))),
-                "experiment_container_position": row.get("Experiment Start Date"),
-             }
-            self.sample_rows.append(sample_row)
-
-            import_result = wipe_import_row_result(import_result, row)
+        row_id = row.get("#", "")
+        self.temporary_experiment_id = row.get("Experiment ID", None)
+        import_result = self.get_row_result_class()()
+        if row_id.isnumeric() and self.temporary_experiment_id: # Uses row ID to identify what are the data rows (compared to title or empty rows)
```

Comment:
While testing, I noticed that isnumeric only works on string. So this may work for the unit tests in csv file, but for the xlsx files that will be uploaded, row_id is actually an int. Therefore, I suggest checking for a condition such as is_row_id_number = isinstance(row_id, int) or (isinstance(row_id, str) and row_id.isnumeric()) for example

Reply:
casting the row_id to str to ensure this works...